src/boundary/Multigrid.cpp

```diff
@@ -697,115 +719,147 @@ Obstacle** Multigrid::obstacleDominantRestriction(size_t level) {
         size_t j2_coarse = (j2_fine + 1) / 2;
         size_t k2_coarse = (k2_fine + 1) / 2;
 
-        // TODO exit?
 #ifndef BENCHMARKING
-        if (i2_fine - i1_fine + 1< domain->get_nx(level - 1) - 2 && i2_coarse - i1_coarse + 1 >= domain->get_nx(level) - 2) {
+        if (i2_fine - i1_fine + 1 < domain->get_nx(level - 1) - 2
+            && i2_coarse - i1_coarse + 1 >= domain->get_nx(level) - 2) {
             m_logger->warn("Be cautious! Obstacle fills up inner cells in x-direction at level {}", level);
         }
-        if (j2_fine - j1_fine +1< domain->get_ny(level - 1) - 2 && j2_coarse - j1_coarse + 1 >= domain->get_ny(level) - 2) {
+        if (j2_fine - j1_fine + 1 < domain->get_ny(level - 1) - 2
+            && j2_coarse - j1_coarse + 1 >= domain->get_ny(level) - 2) {
             m_logger->warn("Be cautious! Obstacle fills up inner cells in y-direction at level {}", level);
         }
-        if (k2_fine - k1_fine +1< domain->get_nz(level - 1) - 2 && k2_coarse - k1_coarse + 1 >= domain->get_nz(level) - 2) {
+        if (k2_fine - k1_fine + 1 < domain->get_nz(level - 1) - 2
+            && k2_coarse - k1_coarse + 1 >= domain->get_nz(level) - 2) {
             m_logger->warn("Be cautious! Obstacle fills up inner cells in z-direction at level {}", level);
         }
+
+        for (size_t c = 0; c < id; c++) {
+            bool overlap = control_obstacle_overlap(obstacleList_coarse[c], &i1_coarse, &i2_coarse,
```

Comment:
this var seem unused?

Reply:
which var? c? it's used. `obstacleList_coarse[c]` though I see, that I forgot to rename the obstacleList to an underscore notation.

Reply:
ah the bool var. yeah, not intentional. just a fail.

Reply:
done in 51a193aef05a7a9850e261de0b6ef6641c2831b4